### PR TITLE
Feature/feature group crud

### PR DESCRIPTION
**Pull request by AlbertPG**

### CODE REVIEW

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/GroupsFragment.java [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+    /**
+     * Create a dialog to delete the group.
+     * @param circularEntryView The entry to which the dialog is associated to
+     * @return The dialog that is associated with the entry
+     */
+    private MaterialAlertDialogBuilder createDeleteGroupDialog(CircularEntryView circularEntryView) {
+        MaterialAlertDialogBuilder dialog = new MaterialAlertDialogBuilder(Objects.requireNonNull(getContext()),
+            R.style.AlertDialogTheme);
+        Group group = (Group) circularEntryView.getObject();
+        dialog.setTitle(getString(R.string.delete_group_title) + " " + group.getName() + INTERROGATION_SIGN);
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/GroupsFragment.java:97]: Codacy Issue found: [The String " " appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5378994)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/CircularEntryView.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/EventView.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrCreateNewGroup.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrCreateNewGroup.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/features/community/Post.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/GroupsView.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/services/CommunityAdapter.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrObtainAllGroups.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrDeleteGroup.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/services/StubCommunityService.java  5
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/GroupsFragment.java  4
- app/src/main/java/org/pesmypetcare/mypetcare/features/community/Group.java  4
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrObtainAllGroups.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/GroupComponentView.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/PetsInfoView.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/features/community/Forum.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrDeleteGroup.java  1
         

Clones added
============
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/EventView.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/PetsInfoView.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5378994&bid=17637427)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/services/CommunityAdapter.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrCreateNewGroup.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrDeleteGroup.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/features/community/Group.java  4
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrObtainAllGroups.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/GroupsView.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/GroupsFragment.java  4
- app/src/main/java/org/pesmypetcare/mypetcare/services/StubCommunityService.java  5
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/CircularEntryView.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrDeleteGroup.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/EventView.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/features/community/Forum.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/PetsInfoView.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/GroupComponentView.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrObtainAllGroups.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/features/community/Post.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrCreateNewGroup.java  2
         

Clones added
============
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/EventView.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/PetsInfoView.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5378994&bid=17637427)

File: app/src/main/java/org/pesmypetcare/mypetcare/features/community/Post.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
+package org.pesmypetcare.mypetcare.features.community;
```
Comment (enrichp) [inline app/src/main/java/org/pesmypetcare/mypetcare/features/community/Post.java:1]: Comments

**Review by xavier-campos-diaz — APPROVED** [review]

**Review by enrichp — APPROVED** [review]

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/features/community/Group.java  4
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrCreateNewGroup.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/GroupsFragment.java  4
- app/src/main/java/org/pesmypetcare/mypetcare/services/CommunityAdapter.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrCreateNewGroup.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/features/community/Post.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrDeleteGroup.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/GroupComponentView.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/EventView.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/PetsInfoView.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrObtainAllGroups.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/features/community/Forum.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrObtainAllGroups.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/GroupsView.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/CircularEntryView.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/services/StubCommunityService.java  5
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrDeleteGroup.java  1
         

Clones added
============
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/EventView.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/PetsInfoView.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5378994&bid=17637427)

**Pull request merged**